sklearnex/tests/test_patching.py

```diff
@@ -116,7 +116,32 @@ def _load_all_models(patched):
 UNPATCHED_MODELS = _load_all_models(patched=False)
 
 
-@pytest.mark.parametrize(("patched", "unpatched"), zip(PATCHED_MODELS, UNPATCHED_MODELS))
-def test_is_patched_instance(patched, unpatched):
+@pytest.mark.parametrize("name", PATCHED_MODELS.keys())
+def test_is_patched_instance(name):
+    patched = PATCHED_MODELS[name]
+    unpatched = UNPATCHED_MODELS[name]
     assert is_patched_instance(patched), f"{patched} is a patched instance"
     assert not is_patched_instance(unpatched), f"{unpatched} is an unpatched instance"
+
+
+@pytest.mark.parametrize("name", PATCHED_MODELS.keys())
+def test_docstring_patching_match(name):
+    patched = PATCHED_MODELS[name]
+    unpatched = UNPATCHED_MODELS[name]
+    patched_docstrings = {
+        i: getattr(patched, i).__doc__
+        for i in dir(patched)
+        if not i.startswith("_") and not i.endswith("_")
+    }
+    unpatched_docstrings = {
+        i: getattr(unpatched, i).__doc__
+        for i in dir(unpatched)
+        if not i.startswith("_") and not i.endswith("_")
+    }
+
+    # check class docstring match
+    assert patched.__doc__ == unpatched.__doc__
```

Comment:
The question is - should they match 1-to-1 or should our documentation be an **extension** over what is in the vanilla sklearn?

Reply:
Because in my mind we can have additional parameters as well as some caveats that should be exposed to the user like the list of conditions for the actual patching

Reply:
I totally agree. I think I may change it to making sure its not 'None', which means that to pass CI it must be set to something, and hopefully then it will spur a conversation.